### PR TITLE
Add changesetBaseRef configuration option for versioning (#1195)

### PR DESCRIPTION
**Pull request by fggrimshaw**
**What's the problem this PR addresses?**

This PR implements issue #1195

**How did you fix it?**

* Add changesetBaseRef, allowing for comparing files against non-master refs

* Update yarnrc documentation accordingly

### CODE REVIEW

File: packages/plugin-version/sources/versionUtils.ts [hunk 1]
```diff
@@ -19,8 +19,8 @@ export enum Decision {
 export type Releases =
   Map<Workspace, Exclude<Decision, Decision.UNDECIDED>>;
 
-export async function fetchBase(root: PortablePath) {
-  const candidateBases = [`master`, `origin/master`, `upstream/master`];
+export async function fetchBase(root: PortablePath, {baseRef}: {baseRef?: string | null}) {
+  const candidateBases = baseRef !== null ? [baseRef] : [`master`, `origin/master`, `upstream/master`];
```
Comment (arcanis) [inline packages/plugin-version/sources/versionUtils.ts:23]: I'm thinking, what do you think about making it an array of refs instead, that would default to `['master', 'origin/master', 'upstream/master']` (what it currently is)? This way people who want to use a custom branch would be able to keep the behaviour that tries to take the origin into account if possible.

Comment (fggrimshaw) [reply packages/plugin-version/sources/versionUtils.ts:23]: I wanted to do that, but did not know how to use an array type. I see now I just need to add `isArray: true`. I will add this

Comment (fggrimshaw) [reply packages/plugin-version/sources/versionUtils.ts]: Hey Maël, I have updated the code and documentation accordingly. I've tested it and it works well. The only limitation is that using the environment variable will only allow a single ref to be set, not a huge deal imo. That could also be user error, I didn't take the time to see if there was a syntax to pass an array.

**Comment by arcanis** [comment]
Thanks!

**Pull request merged**